vibi-dpu/src/core/coverage.rs

```diff
@@ -25,20 +25,46 @@ pub async fn process_coverage(hunkmap: &HunkMap, review: &Review, repo_config: &
                 let comment = comment_text(coverage_map, repo_config.auto_assign());
                 // add comment
-                add_comment(&comment, review, &access_token).await; 
+                add_comment(&comment, review, &access_token).await;
             }
             if repo_config.auto_assign() {
-                // add reviewers
-                println!("Adding reviewers...");
                 let mut author_set: HashSet<String> = HashSet::new();
                 author_set.insert(prhunk.author().to_string());
                 for blame in prhunk.blamevec() {
-                    if author_set.contains(blame.author()) {
+                    let blame_author_opt = author_from_commit(blame.commit(),
+                        hunkmap.repo_name(), hunkmap.repo_owner()).await;
+                    if blame_author_opt.is_none() {
+                        eprintln!("[process_coverage] Unable to get blame author for commit: {}", &blame.commit());
                         continue;
                     }
-                    author_set.insert(blame.author().to_string());
-                    let author_id = blame.author();
-                    add_reviewers(blame.author(), review, &access_token).await;
+                    let blame_author = blame_author_opt.expect("Empty blame_author_opt");
+                    let user_key = blame_author.display_name().to_string();
+                    let blame_user_opt = get_workspace_user_from_db(&user_key);
+                    if blame_user_opt.is_none() {
+                        eprintln!("[process_coverage] no user found in db for blame author: {:?}", blame_author);
+                        continue;
+                    }
+                    let blame_user = blame_user_opt.expect("Empty blame user");
+                    let bb_blame_user_opt = blame_user.users().iter().find(|user| user.uuid() == blame_author.uuid());
+                    let bb_blame_user: BitbucketUser;
+                    if bb_blame_user_opt.is_none() {
+                        let bb_users: Vec<BitbucketUser> = blame_user.users().iter().cloned().collect();
+                        let user_opt = bb_users.get(0);
+                        if user_opt.is_none() {
+                            eprintln!("[process_coverage] Empty Bitbucket Workspace User: {:?}", &blame_user);
+                            continue;
+                        }
+                        bb_blame_user = user_opt.expect("Empty user_opt").to_owned();
+                    }
+                    else {
+                        bb_blame_user = bb_blame_user_opt.expect("Empty bb_blame_user_opt").to_owned();
+                    }
+                    let author_uuid = bb_blame_user.uuid();
+                    if author_set.contains(author_uuid) {
+                        continue;
+                    }
+                    author_set.insert(author_uuid.to_string());
+                    add_reviewers(&blame_user, review, &access_token).await;
```

Comment:
This adds all the users with the same display name as reviewers regardless of whether you were able to find the user in the bitbucket workspace or not.